sdk/typescript/entrypoint/load.ts

```diff
@@ -11,41 +13,102 @@ export async function load(files: string[]): Promise<void> {
-export async function loadArg(value: string): Promise<any> {
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  let parsedValue: any
-
-  const isString = (): boolean =>
-    (value.startsWith('"') && value.endsWith('"')) ||
-    (value.startsWith(`'`) && value.endsWith(`'`))
-  const isArray = (): boolean => value.startsWith("[") && value.endsWith("]")
-
-  // Apply JSON parse to parse array or string if the value is wrapped into a string or array
-  if (isString() || isArray()) {
-    parsedValue = JSON.parse(value)
-  } else {
-    parsedValue = value
+export function loadArgType(
+  scanResult: ScanResult,
+  parentName: string,
+  fnName: string,
+  argName: string
+): TypeDef<TypeDefKind> {
+  const classTypeDef = scanResult.classes.find((c) => c.name === parentName)
+  if (!classTypeDef) {
+    throw new Error(`could not find class ${parentName}`)
+  }
+
+  const methodTypeDef = classTypeDef.methods.find((m) => m.name === fnName)
+  if (!methodTypeDef) {
+    throw new Error(`could not find method ${fnName}`)
+  }
+
+  const argTypeDef = methodTypeDef.args.find((a) => a.name === argName)
+  if (!argTypeDef) {
+    throw new Error(`could not find argument ${argName} type`)
+  }
+
+  return argTypeDef.typeDef
+}
+
+/**
+ * Load the property type from the scan result.
+ *
+ * @param scanResult Result of the scan
+ * @param parentName Class called
+ * @param propertyName property of the class
+ * @returns the type of the property
+ */
+export function loadPropertyType(
+  scanResult: ScanResult,
+  parentName: string,
+  propertyName: string
+): TypeDef<TypeDefKind> {
+  const classTypeDef = scanResult.classes.find((c) => c.name === parentName)
+  if (!classTypeDef) {
+    throw new Error(`could not find class ${parentName}`)
+  }
+
+  const propertyTypeDef = classTypeDef.fields.find(
+    (p) => p.name === propertyName
+  )
+  if (!propertyTypeDef) {
+    throw new Error(`could not find property ${propertyName} type`)
   }
 
-  // If it's a string, it might contain an identifier to load, or it might be a hidden array
-  if (typeof parsedValue === "string") {
-    const [source] = parsedValue.split(":")
+  return propertyTypeDef.typeDef
+}
+
+/**
+ * This function load the argument as a Dagger type.
+ *
+ * Note: The JSON.parse() is required to remove extra quotes
+ */
+export async function loadArg(
+  value: string,
+  type: TypeDef<TypeDefKind>
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+): Promise<any> {
+  switch (type.kind) {
+    case TypeDefKind.ListKind:
+      return JSON.parse(value)
+    case TypeDefKind.ObjectKind: {
+      const objectType = (type as TypeDef<TypeDefKind.ObjectKind>).name
+
+      // This ID might be wrapped in quotes depending on the source.
+      // We need to remove them to be able to call the loading function.
+      if (value.startsWith('"') && value.endsWith('"')) {
+        value = JSON.parse(value)
+      }
```

Comment:
How can I know if an object is idable?

Reply:
We've synced on this, just FTR; the problem was passing to `loadArg` decoded JSON from the parent value but encoded JSON from function arg values. So, it needs decoding in the latter case as well so that `loadArg` only deals with decoded JSON.

> How can I know if an object is idable?

A short and simple answer is to check if there's a `dag[load{objName}FromID]`.